mlos_core/mlos_core/optimizers/optimizer.py

```diff
@@ -120,7 +120,7 @@ def _register(self, *, configs: pd.DataFrame, scores: pd.DataFrame,
         """
         pass    # pylint: disable=unnecessary-pass # pragma: no cover
 
-    def suggest(self, *, context: Optional[pd.DataFrame] = None, defaults: bool = False) -> pd.DataFrame:
+    def suggest(self, *, context: Optional[pd.DataFrame] = None, defaults: bool = False) -> Tuple[pd.DataFrame, Optional[pd.DataFrame]]:
```

Comment:
Mostly looks ok, though am wondering if we should have this `Tuple[DataFrame, DataFrame]` a named data structure as well.

Reply:
I thought about this, but if we make that change, it has to make a quite significant change to the overall API, since then how you call register should change too, and

Reply:
Yeah, I think that's where I landed as well.  We can always tweak it later.